backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/mapper/converter/SpecificationConverter.kt

```diff
@@ -24,19 +25,26 @@ class SpecificationConverter constructor(
         mappingContext: MappingContext?
     ): Specification {
         var fileContent = specificationFileDto.fileContent ?: ""
+        val metaData = specificationFileDto.metaData
+        val isGraphQLFile = metaData != null
 
         if (specificationFileDto.fileUrl != null && specificationFileDto.fileUrl != "") {
             fileContent = specificationFileService.retrieveFile(specificationFileDto.fileUrl)
         }
 
-        val content = specificationDataService.parseFileContent(fileContent)
+        val content = if (isGraphQLFile) fileContent else specificationDataService.parseFileContent(fileContent)
         val uuid = specificationFileDto.id ?: UUID.randomUUID()
 
+        val title = metaData?.title?: specificationDataService.readTitle(content)
```

Comment:
How about we save this data for OpenApi too? Then we wouldn't have to do this here.